homeassistant/components/reolink/select.py

```diff
@@ -0,0 +1,123 @@
+"""Component providing support for Reolink select entities."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any
+
+from reolink_aio.api import DayNightEnum, Host, SpotlightModeEnum
+
+from homeassistant.components.select import SelectEntity, SelectEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ReolinkData
+from .const import DOMAIN
+from .entity import ReolinkCoordinatorEntity
+
+
+@dataclass
+class ReolinkSelectEntityDescriptionMixin:
+    """Mixin values for Reolink select entities."""
+
+    method: Callable[[Host, int, str], Any]
+    get_options: list[str] | Callable[[Host, int], list[str]]
+
+
+@dataclass
+class ReolinkSelectEntityDescription(
+    SelectEntityDescription, ReolinkSelectEntityDescriptionMixin
+):
+    """A class that describes select entities."""
+
+    supported: Callable[[Host, int], bool] = lambda api, ch: True
+    value: Callable[[Host, int], str] | None = None
+
+
+SELECT_ENTITIES = (
+    ReolinkSelectEntityDescription(
+        key="floodlight_mode",
+        name="Floodlight mode",
+        icon="mdi:spotlight-beam",
+        entity_category=EntityCategory.CONFIG,
+        translation_key="floodlight_mode",
+        get_options=[mode.name for mode in SpotlightModeEnum],
+        supported=lambda api, ch: api.supported(ch, "floodLight"),
+        value=lambda api, ch: SpotlightModeEnum(api.whiteled_mode(ch)).name,
+        method=lambda api, ch, name: api.set_whiteled(ch, mode=name),
+    ),
+    ReolinkSelectEntityDescription(
+        key="day_night_mode",
+        name="Day night mode",
+        icon="mdi:theme-light-dark",
+        entity_category=EntityCategory.CONFIG,
+        translation_key="day_night_mode",
+        get_options=[mode.name for mode in DayNightEnum],
+        supported=lambda api, ch: api.supported(ch, "dayNight"),
+        value=lambda api, ch: DayNightEnum(api.daynight_state(ch)).name,
+        method=lambda api, ch, name: api.set_daynight(ch, DayNightEnum[name].value),
+    ),
+    ReolinkSelectEntityDescription(
+        key="ptz_preset",
+        name="PTZ preset",
+        icon="mdi:pan",
+        get_options=lambda api, ch: list(api.ptz_presets(ch).keys()),
```

Comment:
Calling `.keys()` should not be needed.

```suggestion
        get_options=lambda api, ch: list(api.ptz_presets(ch)),
```